### PR TITLE
task_52021_view_result

### PR DESCRIPTION
**Pull request by TMHuy01-08**
## Related Tickets
- [##52021](https://edu-redmine.sun-asterisk.vn/issues/52021)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- Người dùng xem lại bài kiểm tra

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by MinhHieu2804** [comment]
approved

**Comment by TMHuy01-08** [comment]
ready

File: app/views/exams/_do_test.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        <%= form.fields_for question, index: question.id do |field| %>
+          <div class= "answer-collect">
+            <%= field.collection_check_boxes :id, question.answers, :id, :content do |answer| %>
+              <%= answer.label { answer.check_box + " " + answer.text } %><br>
+            <% end %>
+          </div>
+        <% end %>
+      <% end %>
+      <br>
+    <%end%>
```
Comment (ngocvt-0484) [inline app/views/exams/_do_test.html.erb]:
```suggestion
    <% end %>
```

File: app/views/exams/_do_test.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="row justify-center">
+      <%= form.submit t(".submit"), class: "btn btn-primary submit-exam col-md-2" %>
+    </div>
+  <% end %>
+<%end%>
```
Comment (ngocvt-0484) [inline app/views/exams/_do_test.html.erb]:
```suggestion
<% end %>
```

File: app/views/exams/_view_test.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<% if object.any? %>
```
Comment (ngocvt-0484) [inline app/views/exams/_view_test.html.erb]: cho cái này ra ngoài partial

File: app/views/exams/show.html.erb [hunk 4]
```diff
@@ -1,40 +1,19 @@
...
-        <br>
-      <%end%>
-      <div class="row justify-center">
-        <%= form.submit t(".submit"), class: "btn btn-primary submit-exam col-md-2" %>
-      </div>
-    <% end %>
-  <%end%>
+  <% if @exam.doing? %>
+    <%= hidden_field_tag :endtime, @exam.endtime %>
+    <%= hidden_field_tag :exam_id, @exam.id%>
```
Comment (ngocvt-0484) [inline app/views/exams/show.html.erb]:
```suggestion
    <%= hidden_field_tag :exam_id, @exam.id %>
```

File: app/views/exams/_do_test.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<% if object.any? %>
+  <%= form_for @exam, url: exam_path(@exam.id), html: {class: "form-horizontal"} do |form| %>
+    <% object.each_with_index do |question, index| %>
+      <span class="question-title"><%= t ".question" %> <%= index + 1 %>: <%=form.label :question_id ,question.question_content %></span>
```
Comment (ngocvt-0484) [inline app/views/exams/_do_test.html.erb]: xuống dòng  khi dòng quá dài

File: app/helpers/exams_helper.rb [hunk 6]
```diff
@@ -8,4 +8,38 @@ def random_list time, max
...
+
+  def result_of_question user_answers, current_question
+    result = ""
+    case current_question.question_type
+    when Question.types[:single_choice]
+      correct_answer = current_question.answers.find_by is_correct: true
+      result = user_answers.include?(correct_answer) ? "correct" : "wrong"
+    when Question.types[:multiple_choice]
+      result = "correct"
+      correct_answers = current_question.answers.where is_correct: true
```
Comment (ngocvt-0484) [inline app/helpers/exams_helper.rb]: cho where vào scope
gọi scope

**Comment by TMHuy01-08** [comment]
ready

**Pull request merged**